client/core/core.go

```diff
@@ -2309,6 +2345,78 @@ func handleRevokeMatchMsg(_ *Core, dc *dexConnection, msg *msgjson.Message) erro
 	return tracker.db.UpdateOrder(metaOrder)
 }
 
+// handleTradeSuspensionMsg is called when a trade suspension notification is received.
+func handleTradeSuspensionMsg(c *Core, dc *dexConnection, msg *msgjson.Message) error {
+	var sp msgjson.TradeSuspension
+	err := msg.Unmarshal(&sp)
+	if err != nil {
+		return fmt.Errorf("trade suspension unmarshal error: %v", err)
+	}
+
+	dc.pendingSuspendsMtx.Lock()
+	defer dc.pendingSuspendsMtx.Unlock()
+
+	// Attempt to cancel and remove pending suspends.
+	if sched := dc.pendingSuspends[sp.MarketID]; sched != nil {
+		if !sched.Stop() {
+			// Drain the channel.
+			<-sched.C
```

Comment:
Sorry I realized you don't need to drain the channel for timers created with AfterFunc.